common/src/autogluon/common/utils/s3_utils.py

```diff
@@ -45,31 +59,87 @@ def download_s3_folder(bucket, prefix, local_path="."):
     This util will download foo to `local_path` and maintain the structure:
         .
         └── local_path/
-            └── foo/
-                ├── test2.txt
+            └── test2.txt/
                 └── temp/
                     └── test3.txt
+                    
+    Parameters
+    ----------
+    bucket: str
+        The name of the bucket
+    prefix: str
+        The prefix of the object/folder to be downloaded
+    local_path: str
+        The local path to download the object/folder into
+    error_if_exists: bool
+        Whether to raise an error if the root folder exists already
+    delete_if_exists: bool
+        Whether to delete the root folder and all contents within if the root folder exists already
+    dry_run: bool
+        Whether to perform the directory creation and file downloading.
+        If True, will isntead log every file that will be downloaded and every directory that will be created
+    verbose: bool
+        Whether to log detailed loggings
```

Comment:
Specify defaults in docstrings when they exist via `error_if_exists: bool, default = True`, ditto for all other instances.